### ISSUE TITLE
Support for Braille Display "BAUM VarioPro 64/80" and associated modules

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5162]
**Reported by Rudolf on 2015-06-18 09:36**
You can contact Michael Willenbacher via m.willenbacher@baum.de to get the correct protocol for the VarioPro display.

**Comment by jcsteh** [comment]
I did receive the protocol documentation for these displays from Baum. Unfortunately, the protocol is quite different to other Baum displays, so I'd need a physical display for at least a month in order to add support for this.

CC @pavel46.

**Comment by leonardder** [comment]
@jcsteh: Would you implement this protocol in the current driver, or would you start an entirely new one?

N.B. Posted on behalf of @babbagecom

**Comment by jcsteh** [comment]
TL;DR: Integrate into the current driver if possible, but let's be flexible if it's too hard.

The protocol follows a similar format, but it uses entirely different opcodes. If it were integrated into the existing driver, there wouldn't be a huge amount of common code, but it also wouldn't conflict with the protocol used by other displays either.

Doing it in the existing driver is probably nicer for users because then we don't have two separate Baum drivers, so I think I'd prefer this if possible. That said, if it were found to be insanely difficult/ugly/unmaintainable to integrate it into the current driver, I wouldn't object to splitting it out.

**Comment by jcsteh** [comment]
FWIW, BRLTTY just has a single driver.

**Comment by feerrenrut** [comment]
P3, While there is demand for this, there is likely to be quite a bit of work in implementing it.

**Comment by Adriani90** [comment]
Here is a scource code which might help:

https://www.nvda-brugergruppen.dk/guides/devDocs/nvda/brailleDisplayDrivers.baum.BrailleDisplayDriver-class.html

**Comment by josephsl** [comment]
Hi, er, @JCSTeh, can we get a confirmation on this? I’d much prefer if we can get a PR going for this. Thanks.

**Comment by leonardder** [comment]
There is, or was, actually a driver in development for this display which is on github. Last time I checked, it didn't yet concider it ready for pr. I will look at it again and will see whether I could get into touch with the author(s).

**Comment by leonardder** [comment]
I realised I haven't yet shared the url to the repository. The beta driver is public.

https://github.com/BAUM-Engineering/VarioPro_NVDA_driver

**Comment by Adriani90** [comment]
Many thanks for sharing. So, what is the next step regarding this driver? Is it any hope that the Baum developer will continue his work on this project, maybe voluntary? Or will it be forged and included in NVDA?

 

 

 

 

Von: Leonard de Ruijter [mailto:notifications@github.com] 
Gesendet: Samstag, 10. Februar 2018 09:47
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Support for Braille Display "BAUM VarioPro 60/80" (#5162)

 

I realised I haven't yet shared the url to the repository. The beta driver is public.

https://github.com/BAUM-Engineering/VarioPro_NVDA_driver

**Comment by Adriani90** [comment]
Hello,

I have tested the driver shared by @Leonardder and I have granted feedback from german users as well. The driver is working with Baum Vario pro 80, but the input keys and other display keys are not working when using Baum Vario Pro 64. We will test it with taso and will come up with more feedback.

**Comment by leonardder** [comment]
I appreciate your commitment to having this driver work as good as possible. I need to point the following things out though:

1. Given the current situation at Baum, it is very unlikely that the people from Baum will work on this driver any time soon.
2. As you know, VarioPro is a huge device. Because of that and because I won't be able to get hold of a VarioPro 64, I'm afraid I can't invest time in developing it any time soon either.
3. As per above points, this driver is and will stay out of scope for auto detection of braille displays (#1271) for now. Note that the auto detection system is plugable though, so non-core drivers will be able to benefit from it as well if properly implemented.
4. I am reluctant to advise implementing this driver or an improved derivative into NVDA core in the near future. It would be good to have @MichaelDCurran's opinion about this, though.

**Comment by michaelDCurran** [comment]
I have not seen this driver, but:
In principle I have no problem with adding new drivers for Baum devices, 
even if they have closed down. Many people own Baum devices and will 
continue to into the future.
However, it is true that we will not be able to rely on Baum for 
development or support, and getting access to one of these devices to 
test the driver is imperative for this driver being completed and 
accepted into NVDA core.
If someone owns or can get access to one of these devices, and they or 
someone else have the knowhow to write braille drivers, and have the 
time to do so, then I'm sure the driver can be fixed and integrated.
  The most important point here is though, who ever works on this *must* 
have access to a device.

**Comment by Adriani90** [comment]
Dear all,

after great support from skilled baum product users, we found out that the vario pro 64 problem raised because of lib usb from BRLTTY installed on that computer. I have also reached the man who was involved in testing this driver during the whole development process. He is interested in doing more tests and also writing a detailed report if needed. Here is his e-mail adress:

<Steffen König> sr@gauki.com

The driver shared by @Leonardder works with Vario Pro 80, Vario Pro 64 and TASO. The telefone modules are supported as well, although there are some small issues regarding output braille.

As of now, we are in process to reach Florin Trutiu, who comes from Romania. He is the main developer of this driver.

So, in case this driver will be implemented in NVDA, it has to be considered that driver updates are possible from time to time.

**Comment by leonardder** [comment]
@Adriani90: thanks for your update. For privacy reasons, I've removed the e-mail address you mentioned from your comment.

**Comment by leonardder** [comment]
Note that, since they are very tightly related and part of the Variopro eco system and protocol, I consider it OK to also discuss the TASO module and other VP modules in this issue. Therefore I closed #7985.

**Comment by Adriani90** [comment]
Dear all, what is needed here to continue the implementation work? Users of every module seem to not have any issues. TASO is working, Baum Vario Pro 80 and 64 is working.

**Comment by Adriani90** [comment]
Some updates on this Topic:The production and Distribution of Baum Braille Display has been overtaken by Visiobraille GmbH since 01.08.2018. Following Braille Displays are included:
* Compact 2 and Compact 2+
* VarioPRO 80
* Vario 340
* VarioUltra 20 and 40

The Website of the Company with contact details to get a test device is

https://www.visiobraille.de/index.php?article_id=1&clang=2

**Comment by leonardder** [comment]
> The production and Distribution of Baum Braille Display has been overtaken by Visiobraille GmbH since 01.08.2018.

Have you already seen a device that has been delivered by this new company?

**Comment by pavel46** [comment]
Small correction:

Compact line are not a braille displays but the reading maschines.

Braille displays are VarioPro, Vario340 and VarioUltra 20 and 40.

**Comment by Adriani90** [comment]
@pavel46 oh thanks for pointing this out. Sorry for the mistake.

**Comment by RudolfWeeber** [comment]
I think, the displays will continue to be available. The new manufacturer Visiobraille has repaired my VarioPro beginning of this year.

**Comment by Adriani90** [comment]
The driver has been updated for NVDA 2019.3.1 as well.
@BAUM-engineering is anyone of you planning to raise a Pull request for this to be included in NVDA? I am sure much more users would benefit out of it.

**Comment by pavel46** [comment]
hi, this is true  that firma BAUM doesn't existing any longer, but the VarioPro displays are prodnuced and supported by it successor firma VisioBraille.  best regardsPavel Kaplan Am 20.01.2025 um 18:56 schrieb Adriani90 ***@***.***>:﻿
Given the Baum company doesn't exist anymore since several years, and the support of this braille display is limited in any circumstance anyway, I am closing as abandoned. If someone has further updates / details, please comment and we can reopen.
